stable-diffusion-webui-images-browser/scripts/images_history.py

```diff
@@ -0,0 +1,330 @@
+    page_index.submit(lambda s: -s, inputs=[turn_page_switch], outputs=[turn_page_switch])
+    renew_page.click(lambda s: -s, inputs=[turn_page_switch], outputs=[turn_page_switch])
+
+    turn_page_switch.change(
+        fn=get_image_page, 
+        inputs=[img_path, page_index, filenames, keyword, sort_by], 
+        outputs=[filenames, page_index, history_gallery, img_file_name, img_file_time, img_file_info, visible_img_num, warning_box]
+    )
+    turn_page_switch.change(fn=None, inputs=[tabname_box], outputs=None, _js="images_history_turnpage")
+    turn_page_switch.change(fn=lambda:(gr.update(visible=False), gr.update(visible=False)), inputs=None, outputs=[delete_panel, button_panel])
+
+    # other funcitons
+    set_index.click(show_image_info, _js="images_history_get_current_img", inputs=[tabname_box, image_index, page_index, filenames], outputs=[img_file_name, img_file_time, image_index, hidden])
+    set_index.click(fn=lambda:(gr.update(visible=True), gr.update(visible=True)), inputs=None, outputs=[delete_panel, button_panel]) 
+    img_file_name.change(fn=lambda : "", inputs=None, outputs=[collected_warning])  
+   
+    hidden.change(fn=modules.extras.run_pnginfo, inputs=[hidden], outputs=[info1, img_file_info, info2])
+
+    try:
+        modules.generation_parameters_copypaste.bind_buttons(send_to_buttons, img_file_name, img_file_info)
+    except:
+        pass
+
+def on_ui_tabs():
+    global num_of_imgs_per_page
+    global loads_files_num
+    num_of_imgs_per_page = int(opts.images_history_page_columns * opts.images_history_page_rows)
+    loads_files_num = int(opts.images_history_pages_perload * num_of_imgs_per_page)
+    with gr.Blocks(analytics_enabled=False) as images_history:
+        with gr.Tabs(elem_id="images_history_tab") as tabs:
+            for tab in tabs_list:
+                with gr.Tab(tab):
+                    with gr.Blocks(analytics_enabled=False) :
+                        create_tab(tab)
+        gr.Checkbox(opts.images_history_preload, elem_id="images_history_preload", visible=False)         
+        gr.Textbox(",".join(tabs_list), elem_id="images_history_tabnames_list", visible=False) 
+    return (images_history , "Image Browser", "images_history"),
+
+def on_ui_settings():
+    section = ('images-history', "Images Browser")
+    shared.opts.add_option("images_history_preload", shared.OptionInfo(False, "Preload images at startup", section=section))
+    shared.opts.add_option("images_record_paths", shared.OptionInfo(True, "Record accessable images directories", section=section))
+    shared.opts.add_option("images_delete_message", shared.OptionInfo(True, "Print image deletion messages to the console", section=section))
+    shared.opts.add_option("images_history_page_columns", shared.OptionInfo(6, "Number of columns on the page", section=section))
+    shared.opts.add_option("images_history_page_rows", shared.OptionInfo(6, "Number of rows on the page", section=section))
```

Comment:
if the user sets either of columns or rows to 0, on L302 num_of_imgs_per_page will be set to 0, and then on L114 you get a division by zero.